### PR TITLE
Linux - only apply IPv6 routing rules when IPv6 is enabled.

### PR DESCRIPTION
**Pull request by pinkisemils**
The route manager adds IPv6 routing rules even if IPv6 is disabled. On some machines, this fails spectacularly. As such, I've changed the code so IPv6 rules are only added if the user has enabled IPv6.

---
This change is [Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2478)

### CODE REVIEW

File: rustfmt.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ use_field_init_shorthand = true
 condense_wildcard_suffixes = true
 normalize_comments = true
 wrap_comments = true
-merge_imports = true
+imports_granularity = "Crate"
```
Comment (faern) [inline rustfmt.toml:7]: Can we keep unrelated stuff outside of the PR please? I actually started looking into changing our formatting style yesterday. So expect a PR on this soon.

Comment (faern) [reply rustfmt.toml:7]: Not only because I'm nitpicky ;) but also because I might want to backport this PR. Becomes harder if there are unrelated changes.

**Review by pinkisemils — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2478)* status: 0 of 7 files reviewed, 1 unresolved discussion (waiting on @faern)

___
*[rustfmt.toml, line 7 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2478#-MTebGPfENcJvCkiPBHf-r1-7:-MTfa1za0XLt14K7cTFf:bm5ay1j) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3a111224c2634a836c890a9145b4bd061b26916d/rustfmt.toml#L7)):*
<details><summary><i>Previously, faern (Linus Färnstrand) wrote…</i></summary><blockquote>

Not only because I'm nitpicky ;) but also because I might want to backport this PR. Becomes harder if there are unrelated changes.
</blockquote></details>

Sure, I'll split it off in a separate one.

**Review by dlon — APPROVED** [review]
> Reviewed 6 of 7 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2478)* status: all files reviewed, 2 unresolved discussions (waiting on @faern and @pinkisemils)

___
*[talpid-core/src/routing/linux.rs, line 168 at r2](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2478#-MTflM4P0xWgnXbCEKS_:-MTflM4Q0j0oYS4qNu2v:bfqm4y2) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/2209b3d3a5d1f62c88ec8a23083a6ce5fbb06c4c/talpid-core/src/routing/linux.rs#L168)):*
> ```XML
>             while let Some(message) = response.next().await {
>                 if let NetlinkPayload::Error(error) = message.payload {
>                     if rule.header.family == AF_INET6 as u8 && error.code == -libc::EAFNOSUPPORT {
> ```

Do we need this? If you've enabled IPv6 in the service but have disabled it in the kernel, isn't it supposed to fail?

**Review by pinkisemils — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2478)* status: all files reviewed, 2 unresolved discussions (waiting on @dlon and @faern)

___
*[talpid-core/src/routing/linux.rs, line 168 at r2](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2478#-MTflM4P0xWgnXbCEKS_:-MTjIJvj4y17w_5Mnlpw:bn87xwg) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/2209b3d3a5d1f62c88ec8a23083a6ce5fbb06c4c/talpid-core/src/routing/linux.rs#L168)):*
<details><summary><i>Previously, dlon (David Lönnhager) wrote…</i></summary><blockquote>

Do we need this? If you've enabled IPv6 in the service but have disabled it in the kernel, isn't it supposed to fail?
</blockquote></details>

I'm torn, technically it'd make this code be more robust, on the other hand, if IPv6 is enabled in the app but disabled in the kernel, the daemon will end up in a blocked state before even attempting to connect to a relay.

**Review by pinkisemils — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2478)* status: all files reviewed, 2 unresolved discussions (waiting on @dlon and @faern)

___
*[talpid-core/src/routing/linux.rs, line 168 at r2](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2478#-MTflM4P0xWgnXbCEKS_:-MTjIZPR-zJ2nMi9Jsh0:b5n4qse) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/2209b3d3a5d1f62c88ec8a23083a6ce5fbb06c4c/talpid-core/src/routing/linux.rs#L168)):*
<details><summary><i>Previously, pinkisemils (Emīls Piņķis) wrote…</i></summary><blockquote>

I'm torn, technically it'd make this code be more robust, on the other hand, if IPv6 is enabled in the app but disabled in the kernel, the daemon will end up in a blocked state before even attempting to connect to a relay.
</blockquote></details>

I'll remove this code.

**Review by pinkisemils — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2478)* status: 4 of 6 files reviewed, 2 unresolved discussions (waiting on @dlon and @faern)

___
*[rustfmt.toml, line 7 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2478#-MTebGPfENcJvCkiPBHf-r1-7:-MTjM5AR-bM0NTAzbH_B:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3a111224c2634a836c890a9145b4bd061b26916d/rustfmt.toml#L7)):*
<details><summary><i>Previously, pinkisemils (Emīls Piņķis) wrote…</i></summary><blockquote>

Sure, I'll split it off in a separate one.
</blockquote></details>

Done.

**Review by faern — COMMENTED** [review]
> Reviewed 1 of 3 files at r3.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2478)* status: 4 of 6 files reviewed, 1 unresolved discussion (waiting on @dlon)

**Review by dlon — APPROVED** [review]
> Reviewed 2 of 3 files at r3.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2478)* status: all files reviewed, 1 unresolved discussion (waiting on @dlon)

**Review by dlon — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2478)* status: :shipit: complete! all files reviewed, all discussions resolved
